src/drawing.rs

```diff
@@ -8,6 +8,46 @@ use std::cmp::{min, max};
 use std::f32;
 use std::i32;
 
+use pixelops::weighted_sum;
+use rusttype::{FontCollection, Scale, point, PositionedGlyph};
+
+pub fn draw_text<I>(image: &mut I, color: I::Pixel, x: u32, y: u32, height: f32, offset: f32, scale: Scale, font_data: &[u8], text: &str) 
+    where I: GenericImage
+{
```

Comment:
The compiler error you're seeing is because you're using weighted_sum, which requires Pixels satisfying `P::Subpixel: ValueInto<f32> + Clamp<f32>`, but this function is missing that constraint.